internal/metrics/metrics.go

```diff
@@ -0,0 +1,39 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+)
+
+func SanitizeCode(s int) string {
+	switch s {
+	case 0, 200:
+		return "200"
+	default:
+		return strconv.Itoa(s)
+	}
+}
+
+// Only support the list of "regular" HTTP methods, see
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+var methodMap = map[string]string{
+	"GET": http.MethodGet, "get": http.MethodGet,
```

Comment:
What if a method comes in as "Get"? (This whole patch is motivated by adversarial inputs, after all.)

I kinda liked the `strings.ToUpper()` as was done previously, since it covers all case differences, and you don't need to repeat each method for different cases.

Reply:
> What if a method comes in as "Get"? (This whole patch is motivated by adversarial inputs, after all.)
> 
> I kinda liked the `strings.ToUpper()` as was done previously, since it covers all case differences, and you don't need to repeat each method for different cases.

It's a fair question, but would be _very_ unusual. This code was inspired by the Prometheus client code (https://github.com/prometheus/client_golang/blob/9075cdf61646b5adf54d3ba77a0e4f6c65cb4fd7/prometheus/promhttp/instrument_server.go#L377), which assumes the same cases (but outputs lowercase instead). A map lookup is also faster than a call to `strings.ToUpper`, which is why I chose to do that instead.

Personally I have no issue using the label `"OTHER"` when the input method is abnormal. I've used the `prometheus/client_golang` code for years, and have never seen legitimate clients using different method casings.

Reply:
Ok; up to you, I just think repeating the method name in the map is a bit odd, and it seems unfortunate to lose the robustness of just uppercasing the method (which is very, very cheap -- utter nanoseconds I bet).